### PR TITLE
Added function to render reprex in a different locale

### PR DESCRIPTION
**Pull request by ellessenne**
Hi,
I added a function named `reprex_locale` that wraps `reprex` and uses `withr::local_envvar` and `withr::local_locale` to modify the locale before rendering the reprex.

For instance, running

```r
reprex_locale({
  dplyr::select(foo)
  }, locale = "it_IT")
```

returns:

``` r
dplyr::select(foo)
#> Error in dplyr::select(foo): oggetto "foo" non trovato
```

<sup>Created on 2019-07-08 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9000)</sup>

The locale modifications are:

1. Setting the environment variable `LANG` to the new locale value;
1. Setting the environment variable `LC_MESSAGES` to the new value.

The latter is responsible for translating messages (and errors, warnings) to a different locale, while the former deals with e.g. the month names:

```r
reprex_locale({
  df <- data.frame(stringsAsFactors = FALSE,
                   date = as.Date(c("2019-01-01", "2019-02-01")),
                   value = c(1, 2))
  plot(df$date, df$value)
}, locale = "es_ES")
```

``` r
df <- data.frame(stringsAsFactors = FALSE,
                 date = as.Date(c("2019-01-01", "2019-02-01")),
                 value = c(1, 2))
plot(df$date, df$value)
```

<sup>Created on 2019-07-08 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9000)</sup>

I haven't touched the RStudio addin yet - I wonder if it would actually be better to have a separate one rather than modifying the current one?

Alessandro

### CODE REVIEW

**Comment by hadley** [comment]
Fixes #250

File: NEWS.md [hunk 1]
```diff
@@ -5,6 +5,8 @@
 * `venue = "so"` (SO = Stack Overflow) has converged with default `venue = "gh"` (GitHub). As of January 2019, SO [supports CommonMark fenced code blocks](https://meta.stackexchange.com/questions/125148/implement-style-fenced-markdown-code-blocks/322000#322000). The only remaining difference is that Stack
 Overflow does not support the collapsible details tag that we use on GitHub for the session info requested via `si = TRUE` (#231).
 
+* Added `reprex_locale` function to render a reprex using a different locale (@ellessenne, #250).
```
Comment (hadley) [inline NEWS.md]:
```suggestion
* Added `reprex_locale()` to render a reprex using a different locale (@ellessenne, #250).
```

File: R/reprex-locale.R [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#' Render a reprex in a different locale
+#'
+#' @description
```
Comment (hadley) [inline R/reprex-locale.R]:
```suggestion
```

**Comment by ellessenne** [comment]
I realised that `LC_MESSAGES` is not supported on Windows, and that it is actually not necessary: setting the environment variable `LANG` *should* be enough?

I'll also deal with the edits @hadley suggested ASAP!

**Comment by jennybc** [comment]
Thanks @ellessenne! It turns out you tackled a fairly hard issue -- harder than we knew. This solution does not work on Windows.

So, with gratitude and a "sorry", I'm afraid I have to close this one unmerged. I am converging on a solution, so thanks for providing the impetus for me to get to the bottom of this.

**Comment by ellessenne** [comment]
Thanks for the feedback @jennybc and of course don't worry - looking forward to checking out the solution you're converging to! 😃

**Comment by jennybc** [comment]
It's here: 2ea429e7c58193351a3b7376c26c399ba47c641e

Definitely a case of very little new code, but took quite a long time to gain enough understanding to write it! The differences between OSes are pretty big and aggravating too.

**Comment by ellessenne** [comment]
That is so interesting, thanks - it doesn't seem trivial at all, especially with all the subtle OS differences!
Thanks again for pointing me to that @jennybc!

**Pull request closed without merging**